src/libraries/System.Net.Http/tests/TrimmingTests/MetricsHandlerTrimmedTest.cs

```diff
@@ -0,0 +1,37 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Runtime.Versioning;
+using System.Threading.Tasks;
+
+class Program
+{
+    static async Task<int> Main(string[] args)
+    {
+        using HttpClient client = new();
+
+        // send a request, but ignore its result
+        try
+        {
+            await client.GetAsync("https://www.microsoft.com");
+        }
+        catch { }
+
+        Type metricsHandler = GetHttpType("System.Net.Http.Metrics.MetricsHandler");
```

Comment:
Shall we add a check against the `System.Net.Http.Metrics.SocketsHttpHandlerMetrics` and the `System.Net.Http.Metrics.ConnectionMetrics`  types here, or we don't care if those are also trimmed?

Reply:
I think I fixed that. 

I hoped for much more and drafted it here.
https://github.com/dotnet/runtime/pull/114636

TplEventSource is stubborn!